pkg/server/commands/batch_check_command.go

```diff
@@ -144,30 +158,38 @@ func (bq *BatchCheckQuery) Execute(ctx context.Context, params *BatchCheckComman
 
 			response, _, err := checkQuery.Execute(ctx, checkParams)
 
-			resultMap.Store(check.GetCorrelationId(), &BatchCheckOutcome{
+			resultMap.Store(key, &BatchCheckOutcome{
 				CheckResponse: response,
 				Err:           err,
 			})
 
+			totalQueryCount.Add(response.GetResolutionMetadata().DatastoreQueryCount)
+
 			return nil
 		})
 	}
 
 	_ = pool.Wait()
 
 	results := map[CorrelationID]*BatchCheckOutcome{}
-	var totalQueryCount uint32
 
-	resultMap.Range(func(k, v interface{}) bool {
-		// Convert types since sync.Map is `any`
-		outcome := v.(*BatchCheckOutcome)
-		results[CorrelationID(k.(string))] = outcome
+	duplicateCount := 0
```

Comment:
Can't this be derived from number of input checks minus number of cache keys?

Reply:
that is a much more clear way to do it, instead of the math currently down in 179. Will change